teps/0059-skip-guarded-task-only.md

```diff
@@ -95,6 +107,28 @@ updates.
 
 This TEP addresses skipping strategies to give users the flexibility to skip a single guarded `Task` only and unblock execution of its dependent `Tasks`.
 
+Today, we support specifying a list of `WhenExpressions` through the `when` field as such:
+
+```yaml
+when:
+  - input: 'foo'
+    operator: in
+    values: [ 'bar' ]
+```
+
+We propose changing the `when` field from a list to a dictionary and adding `scope` and `expressions` fields under the `when` field.
+- The `scope` field would be used to specify whether the `WhenExpressions` guard the `Task` only or the whole `Branch`.
```

Comment:
NIT: it might be worth adding that the `Branch` scope matches the current behaviour?

Reply:
added, thanks :)